README.md

```diff
@@ -182,6 +182,102 @@ You can intermittently poll whether the operation has finished by using the `Pol
+
+If you do NOT want a newly created `Context` to be a child of `ApplicationContext`, use the `Azure::Core::Context::CreateNewRoot()` function, 
+which will create a new root context.
+
+When a context is copied from another context, the copied context will share state with the original context. 
+This means that if the original context is cancelled, the copied context will also be cancelled.
+
+Cancellation of a `Context` is a permanent operation. Once a non-root context is cancelled, it cannot be un-cancelled.
+
+Root `Contexts` are unique in that their cancellation *CAN* be reset. Cancelling a root context essentially creates a new root Context replacing the existing root.
+
+The following snippet shows creating a new context from the `ApplicationContext` and cancelling the root context..
+
+```cpp
+  // Create a child context from the root context
+  Azure::Core::Context childContext = Azure::Core::Context::ApplicationContext.WithDeadline(std::chrono::system_clock::now() + std::chrono::seconds(10));
+
+  // Check if the child context is cancelled - this will not be printed unless 10 seconds have passed from the creation of childContext.
+  if (childContext.IsCancelled())
+  {
+	std::cout << "Child context is cancelled" << std::endl;
+  }
+
+  // Cancel the root context
+  Azure::Core::Context::ApplicationContext.Cancel();
+
+  // Check if the child context is cancelled - this will be printed because the root context was cancelled.
+  if (childContext.IsCancelled())
+  {
+	std::cout << "Child context is cancelled" << std::endl;
+  }
+
+  // Reset the root context
+  Azure::Core::Context::ApplicationContext.Reset();
+
+  // Check the child context. This will be printed because even though the root context was reset, existing contexts created before
+  // the reset will still be cancelled.
+  if (childContext.IsCancelled())
+  {
+    std::cout << "Child context is still cancelled" << std::endl;
+  }
+```
+
+When a `Context` is cancelled, it is typically indicated by throwing `Azure::Core::OperationCancelledException`. This exception can be caught and handled by the application.
+
+#### Public, Private and Internal Types
+
+The APIs defined in the Azure SDK for C++ fall into three rough categories:
+
+- **Public**: These are the types that are intended to be used by the consumers of the SDK.
+They are part of the public API and are stable. Breaking changes to these types will be avoided as much
+as possible. All public types and functions are located are in the `Azure` namespace.
+- **Internal**: These are the types that are used internally by the packages within the SDK.
+They are intended for use by the packages which make up the Azure SDK. They are NOT intended to be used by the consumers of the SDK.
+Breaking changes to these types are allowed, within certain constraints. These types are located in an `Azure` namespace within the `_internal` terminal namespace (for instance, `Azure::Core::Http::Policies::_internal::RequestActivityPolicy`).
```

Comment:
I use visual studio code, and it likes to reformat lines.

Also, I have this annoying tendency of wanting to be able to read all the text in the document, so I tend to reformat text so that the words fit on my screen.